Add more logging to email sending

Shriukan33 opened the pull request:
We got invalid token in GET request on press mailing list
In an attempt to reproduce the error, we log more details about
it.
Adding the token + email and ID to logs will allow us to better
reproduce the conditions that lead to incorrect token.

JeffAbrahamson commented on transport_nantes/asso_tn/utils.py:
```diff
@@ -96,8 +96,12 @@ def token_valid(encoded_timed_token, test_value_now=None):
-    except (TypeError, ValueError):
-        logger.info(f"token_valid: invalid token ({encoded_timed_token})")
+    except (TypeError, ValueError) as e:
+        # We got invalid token in GET request on press mailing list
```
Please cite the issue and provide a date so that future readers have context.

JeffAbrahamson commented on transport_nantes/topicblog/views.py:
```diff
@@ -459,6 +460,9 @@ def _set_email_context(
         context["token"] = \
             self.get_unsubscribe_token(recipient[0], send_record_id)
 
+        logger.info(f"Token {context['token']} is associated "
```
Please cite the issue and provide a date so that future readers have context.

JeffAbrahamson reviewed (changes requested):
This isn't a fix, it's the beginning of a strategy to understand more about a certain bug for the next time it happens.  As such, you not only need to explain why you do what you do, but you need to make sure that future readers of the code have a quick link to the issue, both from the code (missing) and from the commit message (strangely, also missing, but this is the citation that will give you a back-link from the issue to the commits, short of doing a `git log --grep`, which also needs the issue tagged).

There's _way_ too much state in your head that isn't in the code comments and isn't in the commit.

JeffAbrahamson commented on transport_nantes/asso_tn/utils.py:
```diff
@@ -96,8 +96,18 @@ def token_valid(encoded_timed_token, test_value_now=None):
+        # send.
+        # In an attempt to reproduce and understand the error, we log more
+        # details about it for the next time it appears
+        # See issue https://github.com/transport-nantes/tn_web/issues/777
```
This is good, but note that that issue doesn't have any useful information in it.

Shriukan33 replied:
Added my debug document to issue

JeffAbrahamson commented on transport_nantes/asso_tn/utils.py:
```diff
@@ -96,8 +96,18 @@ def token_valid(encoded_timed_token, test_value_now=None):
+        # of 24 were invalid. We know this because we got 10 times a
+        # "token_valid: invalid token" error on GET requests to the
+        # 'mailing_list:press_subscription_management' url, seconds after the
+        # send.
```
Good.  You might note all the things you learned yesterday whilst debugging this.  For example, that the error occurred in `token_valid()`, whatever else you learned.

You spent (if I understood correctly) over an hour on this.  Surely you learned more than you can summarise in two sentences!

Some of that summary can go in the issue, of course.  The code comment needs to get future devs up to speed on what the issue is, the issue can provide all the gory detail.

Shriukan33 replied:
Reworded + added document to issue

Shriukan33 commented on transport_nantes/asso_tn/utils.py:
```diff
@@ -96,8 +96,18 @@ def token_valid(encoded_timed_token, test_value_now=None):
-    except (TypeError, ValueError):
-        logger.info(f"token_valid: invalid token ({encoded_timed_token})")
+    except (TypeError, ValueError) as e:
+        # On the 31/05/2022 :
```
Reworded

JeffAbrahamson commented:
Minor issues:
* The exceptions don't have spaces in the names: `UnicodeError` rather than `Unicode Error`, etc.
* Blank lines and indents make for more readable comments. ;-)

Otherwise, LGTM.

Shriukan33 commented:
> Minor issues:
> 
>     * The exceptions don't have spaces in the names: `UnicodeError` rather than `Unicode Error`, etc.
> 
>     * Blank lines and indents make for more readable comments. ;-)
> 
> 
> Otherwise, LGTM.

Added some air in the comments, and removed space in error names :)

Shriukan33 commented:
CodeQL check fail is not very helpful, it seems that the result analysis failed to do a PUT request to upload its results.
It might be all on GitHub's end. 

Given the nature of the PR, I doubt they would flag anything anyway, can I merge ?

JeffAbrahamson reviewed (approved):
LGTM.
